File: frontend/projects/shared/src/lib/interfaces/topicNoteDto.ts
export interface TopicNoteDto {
  id: number;
  title: string;
  description: string;
  createdAt: Date;
  pdfUrl: string;
  uploaderName?: string | null;
  teacherId?: number | null;
  teacherName?: string | null;
  subjectId?: number | null;
  topicPoolId?: number | null;
  fileName?: string;
  url?: string;
  publicUrl?: string;
}

File: backend/src/main/java/at/learnhub/dto/TopicNoteDto.java
package at.learnhub.dto;

import java.time.LocalDate;

public class TopicNoteDto {
    public Long id;
    public String title;
    public String description;
    public LocalDate date;

    public String pdfUrl;
    public String uploaderName;
    public Long teacherId;
    public String teacherName;

    public Long subjectId;
    public Long topicPoolId;
}